src/createXcmTypes/util/getAssetId.ts

```diff
@@ -0,0 +1,168 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+
+import { ApiPromise } from '@polkadot/api';
+
+import { ASSET_HUB_CHAIN_ID } from '../../consts';
+import { BaseError, BaseErrorsEnum } from '../../errors';
+import { Registry } from '../../registry';
+import { foreignAssetMultiLocationIsInCacheOrRegistry } from './foreignAssetMultiLocationIsInCacheOrRegistry';
+import { foreignAssetsMultiLocationExists } from './foreignAssetsMultiLocationExists';
+import { getChainIdBySpecName } from './getChainIdBySpecName';
+
+/**
+ *
+ * Given a valid token symbol or MultiLocation, returns it's corresponding integer or MultiLocation assetId
+ * Errors if given an input that is not in the asset registry or not found in either
+ * AssetHub's or the current Parachain's chain state.
+ *
+ * @param api ApiPromise
+ * @param registry Registry
+ * @param asset string
+ * @param specName string
+ * @param isForeignAssetsTransfer boolean
+ */
+export const getAssetId = async (
+	_api: ApiPromise,
+	registry: Registry,
+	asset: string,
+	specName: string,
+	isForeignAssetsTransfer?: boolean
```

Comment:
Im assuming its not applicable here?

Reply:
yes, I don't think we've passed `isLiquidPoolAssetTransfer` as of now. I think because they are always expected to be integers corresponding to the lpToken integer asset ID

Reply:
Makes perfect sense thanks :)